neo4j/_async_compat/concurrency.py

```diff
@@ -423,6 +436,8 @@ def notify_all(self):
         self.notify(len(self._waiters))
 
 
-Condition = threading.Condition
-CooperativeLock = Lock = threading.Lock
-CooperativeRLock = RLock = threading.RLock
+Condition: te.TypeAlias = threading.Condition
+CooperativeLock: te.TypeAlias = threading.Lock
+Lock: te.TypeAlias = threading.Lock
+CooperativeRLock: te.TypeAlias = threading.RLock
```

Comment:
you're not using this or the one below, or did I miss something?

Reply:
`AsyncCooperativeRLock` (and with it `CooperativeRLock`) are actually being used in the Pool implementation
https://github.com/neo4j/neo4j-python-driver/blob/bd5e6a97cd3737f6c4cbe1503d3cc5ccc50fa0ab/neo4j/_async/io/_pool.py#L81
https://github.com/neo4j/neo4j-python-driver/blob/bd5e6a97cd3737f6c4cbe1503d3cc5ccc50fa0ab/neo4j/_sync/io/_pool.py#L81

As is the `(Async)RLock`
https://github.com/neo4j/neo4j-python-driver/blob/bd5e6a97cd3737f6c4cbe1503d3cc5ccc50fa0ab/neo4j/_async/io/_pool.py#L453
https://github.com/neo4j/neo4j-python-driver/blob/bd5e6a97cd3737f6c4cbe1503d3cc5ccc50fa0ab/neo4j/_sync/io/_pool.py#L453